### PR TITLE
Fixes #15351: Documentation for agent key management

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/15351

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: src/api_v2/nodes.md [hunk 1]
```diff
@@ -610,45 +610,49 @@ HTTP/1.1 200 OK
...
     /**
-    @api {post} /api/nodes/{id} 6. Set Node properties
+    @api {post} /api/nodes/{id} 6. Update Node settings and properties
     @apiVersion 8.0.0
     @apiName updateNode
     @apiGroup Nodes
 
     @apiUse nodeId
-    @apiParam (Multi) {Property}       properties A property of the Node defined by its name and its value.
+    @apiParam (Multi) {Property}       properties A property of the Node defined by its name and its value. Value are non-empty UTF-8 strings. Setting a key to the empty string removes it from the list of properties for that node.
```
Comment (amousset) [inline src/api_v2/nodes.md]: Values

File: src/api_v2/nodes.md [hunk 2]
```diff
@@ -610,45 +610,49 @@ HTTP/1.1 200 OK
...
+    @apiParam {PolicyMode}       policy In which mode the node will apply its configuration policy. You can choose between "audit" and "enfoce" modes, or "default" to use the global mode
```
Comment (amousset) [inline src/api_v2/nodes.md]: enforce

File: src/api_v2/nodes.md [hunk 3]
```diff
@@ -610,45 +610,49 @@ HTTP/1.1 200 OK
...
+    @apiParam {State}        the node lifecycle state. ou can choose between "enabled", "ignored", "empty-policies", "initializing", and "preparing-eol" state. See https://docs.rudder.io/reference/current/usage/advanced_node_management.html#node-lifecycle for more information.
```
Comment (amousset) [inline src/api_v2/nodes.md]: ou -> You

File: src/api_v2/nodes.md [hunk 4]
```diff
@@ -666,19 +670,60 @@ HTTP/1.1 200 OK
...
+    @api {post} /api/nodes/{id} 6. Update Node settings and properties
+    @apiVersion 12.0.0
     @apiName updateNode
     @apiGroup Nodes
 
     @apiUse nodeId
     @apiParam (Multi) {Property}       properties A property of the Node defined by its name and its value. Value are non-empty UTF-8 strings. Setting a key to the empty string removes it from the list of properties for that node.
     @apiParam {PolicyMode}       policy In which mode the node will apply its configuration policy. You can choose between "audit" and "enfoce" modes, or "default" to use the global mode
+    @apiParam {State}        the node lifecycle state. ou can choose between "enabled", "ignored", "empty-policies", "initializing", and "preparing-eol" state. See https://docs.rudder.io/reference/current/usage/advanced_node_management.html#node-lifecycle for more information.
+    @apiParam {agentKey.status}   The agent key status (optionnal). Can be "certified" or "undefined".
```
Comment (amousset) [inline src/api_v2/nodes.md]: optional

File: src/api_v2/nodes.md [hunk 5]
```diff
@@ -666,19 +670,60 @@ HTTP/1.1 200 OK
...
+    @apiParam {agentKey.value}    The agent key value (optionnal) in PEM format (either certificate or public key).
```
Comment (amousset) [inline src/api_v2/nodes.md]: optional

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Pull request merged**